pages/api/predictions/index.js

```diff
@@ -6,36 +7,43 @@ const replicate = new Replicate({
   userAgent: `${packageData.name}/${packageData.version}`,
 });
 
+async function getObjectFromRequestBodyStream(body) {
```

Comment:
This works but it doesn't feel right. What's a better way to do this?

Update `replicate` to accept a ReadableStream as `input`?

Reply:
I think `const input = await req.body.json()` could work. Could you try that?

Reply:
```
error - req.body.json is not a function
```

Reply:
Ah well, thanks for trying that out. As cumbersome as this is, all of the other solutions I've found are even worse, somehow.

As far as putting this in the client, I'd imagine that most consumers wouldn't be passing full request bodies directly, so accepting a `ReadableStream` probably wouldn't be helpful.

Reply:
I opened an issue on the Next.js repo for guidance about best practice here: https://github.com/vercel/next.js/issues/48403

In the meantime I'll go ahead and ship this and we can come back if we discover a better approach.